packages/jazz-tools/src/coValues/coMap.ts

```diff
@@ -610,6 +610,20 @@ const CoMapProxyHandler: ProxyHandler<CoMap> = {
             }
         }
     },
+    has(target, key) {
+        if (key in target) {
+            return true;
+        }
+
+        const descriptor = (target._schema[key as keyof CoMap["_schema"]] ||
+            target._schema[ItemsSym]) as Schema;
+
+        if (typeof key === "string" && descriptor) {
+            return target._raw.get(key) !== undefined;
```

Comment:
"in" apparently returns true if an object has a property, even if it's set to undefined. I think the closest we can get to this is seeing if it's in the schema, or in a record like comap if it is defined. Thoughts?

Reply:
hmmm. for my usecases I use it to discriminate the type of an union

Reply:
Ok let's keep it like this for now and see if it ever causes problems